go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -508,15 +508,15 @@
   },
   {
     "comment": "join with unqualified reference optimize routes when source & reference have different names",
-    "query": "select user.col from user join ref_in_source",
+    "query": "select user.col from user join source_of_ref",
```

Comment:
A result of some tests tweaks unrelated to the meaningful changes in the PR.

Reply:
here you are changing the test and not the output. this should be reverted.

Reply:
I see `ref_in_source` is now `ref_in_source`